apps/rush-lib/src/cli/logic/pnpm/PnpmShrinkwrapFile.ts

```diff
@@ -95,6 +95,11 @@ export function extractVersionFromPnpmVersionSpecifier(version: string): string
     if (versionParts.length === 5 && isScoped) {
       extractedVersion = versionParts[3]; // e.g. "3.1.1"
     }
+
+    // e.g. "path.pkgs.visualstudio.com/@scope/depame/1.4.0"
+    if (!extractedVersion && versionParts[versionParts.length - 1].match(/^(\d*)\.(\d*)\.(\d*)/)) {
```

Comment:
>(\d*)\.(\d*)\.(\d*) [](start = 75, length = 19)

Are release tags supported?  For example something like "1.2.3-beta.3"

Reply:
Take a look at [PR #607](https://github.com/Microsoft/web-build-tools/pull/607) which has an example of these versions not getting handled correctly.  The regular expression used for the schemas in that PR was like this:

`"^[0-9]+\\.[0-9]+\\.[0-9a-zA-Z.+\\-]+$"`

We chose to match a superset of valid strings to avoid an overcomplicated pattern

---
In reply to: [182993892](https://github.com/Microsoft/web-build-tools/pull/615#discussion_r182993892) [](ancestors = 182993892)

Reply:
Thank you for the info. I looked more into it and I think per [https://docs.npmjs.com/cli/version](https://docs.npmjs.com/cli/version) the version should satisfy semver.valid so I am using that instead of the regular expression now. It. III

---
In reply to: [183127330](https://github.com/Microsoft/web-build-tools/pull/615#discussion_r183127330) [](ancestors = 183127330,182993892)